server/src/main/java/io/deephaven/server/notebook/FilesystemStorageServiceGrpcImpl.java

```diff
@@ -0,0 +1,188 @@
+package io.deephaven.server.notebook;
+
+import com.google.protobuf.ByteString;
+import com.google.rpc.Code;
+import io.deephaven.configuration.Configuration;
+import io.deephaven.extensions.barrage.util.GrpcUtil;
+import io.deephaven.internal.log.LoggerFactory;
+import io.deephaven.io.logger.Logger;
+import io.deephaven.proto.backplane.grpc.CreateDirectoryRequest;
+import io.deephaven.proto.backplane.grpc.CreateDirectoryResponse;
+import io.deephaven.proto.backplane.grpc.DeleteItemRequest;
+import io.deephaven.proto.backplane.grpc.DeleteItemResponse;
+import io.deephaven.proto.backplane.grpc.FetchFileRequest;
+import io.deephaven.proto.backplane.grpc.FetchFileResponse;
+import io.deephaven.proto.backplane.grpc.FileInfo;
+import io.deephaven.proto.backplane.grpc.FileKind;
+import io.deephaven.proto.backplane.grpc.ListItemsRequest;
+import io.deephaven.proto.backplane.grpc.ListItemsResponse;
+import io.deephaven.proto.backplane.grpc.MoveItemRequest;
+import io.deephaven.proto.backplane.grpc.MoveItemResponse;
+import io.deephaven.proto.backplane.grpc.SaveFileRequest;
+import io.deephaven.proto.backplane.grpc.SaveFileResponse;
+import io.deephaven.proto.backplane.grpc.StorageServiceGrpc;
+import io.grpc.stub.StreamObserver;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Optional;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Stream;
+
+@Singleton
+public class FilesystemStorageServiceGrpcImpl extends StorageServiceGrpc.StorageServiceImplBase {
```

Comment:
I feel that this impl should at least ensure that there exists a session for each of these calls. The `getSession` method already throws a StatusRuntimeException; it's sufficient to fetch the session and not use it.